src/helper/RedBlackNode.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace Src\helper;
+
+use Src\helper\enums\Color;
```

Comment:
PSR4 Violation 
should be

use Src\Helper\Enums\Color

Außerdem ist mir persönlich Color etwas zu generisch. Es sollte RedBlackTreeColor sein, damit es später keine Überschneidungen beim Enum Color gibt.

Reply:
After renaming the `enums` folder PHPStan was complaining that the case was wrong. Was fixed by changing it to singular.

The Helper folder should be renamed in a seperate PR since it effect the code project wide. Open Issue for that.

Reply:
The Color enum was kept generic for potential reuse later on since it could be extended without effecting the usage in the RedBlackNode/Tree implementation.

Gibt es hier ein Argument für die Spezifizierung? Allgemein wäre doch ein wiederverwendbares `Color` Enum sinnvoller als die Spezifizierung eine `RedBlackTreeColor` welches nur einen Zweck erfüllt oder?